File: comgongdaeng/Implementation/BOJ_17413.py
# 스택 사용
# 여는 태그를 만나기 전까지 스택에 push 했던 문자들을 pop
# 여는 태그 ~ 닫는 태그를 만나기 전까지 문자들은 그대로 append

chars = input()
st = []
ans = ''
# 여는 태그를 만났을 때 문자열 뒤집지 않기 위한 변수 선언
tag = False

for char in chars:
    if char == ' ':
        while st:
            ans += st.pop()
        ans += char
    
    elif char == '<':
        while st:
            ans += st.pop()
        ans += char
        tag = True
    elif char == '>':
        ans += char
        tag = False
    elif tag:
        ans += char
    else: st.append(char)

# 마지막으로 스택 비우기
while st:
    ans += st.pop()
print(ans)

File: naarang/Implementation/BOJ_17413.js
const fs = require("fs");
let input = fs
  .readFileSync("naarang/DataStructure/input.txt")
  .toString()
  .trim()
  .split("\n");

let tag = false;
let answer = [];
let word = "";

for (let i of input[0]) {
  if (i === "<" && !tag) {
    tag = true;
    if (word.length > 0) {
      answer.push(word.split("").reverse().join(""));
    }
    word = i;
  } else if (i === ">" && tag) {
    tag = false;
    word += i;
    answer.push(word);
    word = "";
  } else if (tag) {
    word += i;
  } else if (i === " ") {
    if (word.length > 0) {
      answer.push(word.split("").reverse().join(""));
      word = "";
      answer.push(" ");
    }
  } else {
    word += i;
  }
}

if (word.length > 0) {
  answer.push(word.split("").reverse().join(""));
}

console.log(answer.join(""));
